jdl/jhipster/deployment-options.js

```diff
@@ -16,82 +16,124 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-const _ = require('lodash');
-
 const DeploymentTypes = {
   DOCKERCOMPOSE: 'docker-compose',
   KUBERNETES: 'kubernetes',
   OPENSHIFT: 'openshift',
   exists: deploymentType => !!deploymentType && !!DeploymentTypes[deploymentType.toUpperCase().replace('-', '')],
 };
 
-const Options = {
-  deploymentType: {
-    dockerCompose: DeploymentTypes.DOCKERCOMPOSE,
-    rancherCompose: DeploymentTypes.RANCHERCOMPOSE,
-    kubernetes: DeploymentTypes.KUBERNETES,
-    openshift: DeploymentTypes.OPENSHIFT,
-  },
-  gatewayType: {
-    springCloudGateway: 'SpringCloudGateway',
-  },
-  monitoring: {
-    no: 'no',
-    prometheus: 'prometheus',
-  },
-  directoryPath: '../',
-  appsFolders: [],
-  clusteredDbApps: [],
-  // adminPassword: 'admin',
-  serviceDiscoveryType: {
-    eureka: 'eureka',
-    consul: 'consul',
-    no: 'no',
-  },
-  dockerRepositoryName: '',
-  dockerPushCommand: 'docker push',
-  // Kubernetes specific
+const kubernetesRelatedOptions = {
   kubernetesNamespace: 'default',
   kubernetesServiceType: {
     loadBalancer: 'LoadBalancer',
     nodePort: 'NodePort',
     ingress: 'Ingress',
   },
+  kubernetesStorageClassName: '',
+  kubernetesUseDynamicStorage: {
+    false: false,
+    true: true,
+  },
   ingressDomain: '',
+  ingressType: {
+    nginx: 'nginx',
+    gke: 'gke',
+  },
   istio: {
     false: false,
     true: true,
   },
-  // openshift specific
+};
+
+const openshiftRelatedOptions = {
   openshiftNamespace: 'default',
+  registryReplicas: {
+    two: 2,
+  },
   storageType: {
     ephemeral: 'ephemeral',
     persistent: 'persistent',
   },
 };
 
-Options.defaults = (deploymentType = Options.deploymentType.dockerCompose) =>
-  _.omitBy(
```

Comment:
I dont remmber why did omitBy here originally, so hope its still works

Reply:
The omit here is useful if you have undefined values, and as the previous code generated them, the omit's role was to clean things up.

Reply:
And thanks for the review :)